instant-distance/src/lib.rs

```diff
@@ -190,23 +267,66 @@ impl<'a, P, V> MapItem<'a, P, V> {
     }
 }
 
+#[derive(Default)]
+#[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
+struct PointStorage {
```

Comment:
Why is this abstraction needed?

If we're going to do this, this should probably be an implementation of a trait (so we can slot in an memmap-backed storage implementation later).